### PR TITLE
Various small changes

### PR DESCRIPTION
**Pull request by swlynch99**
This is few small changes to fix some pain points I ran into while working through the workshop:
- I have added a small test suite that should work for just about any correct sieve cache implementation. I have also added a note to the end of the `sieve_cache` readme that points to the test suite.
- I have fixed all existing warnings from clippy and otherwise.

### CODE REVIEW

File: example_sieve_cache/src/sieve_cache.rs [hunk 1]
```diff
@@ -85,6 +85,16 @@ where
     }
 }
 
+impl<Key, Value> Default for SieveCache<Key, Value>
```
Comment (kvc0) [inline example_sieve_cache/src/sieve_cache.rs:94]:
```suggestion
/// Implementing Default for a struct is a Rust-ism that basically means "default constructor"
/// if you're coming from Java or similar.
/// You will often see things like `SieveCache::default()`, `Default::default()`, and
/// `#[derive(Default)]`. The first 2 are ways to call this default() function, and the 3rd is a
/// macro that declares your type as being composed of Default-types.
/// It's good to implement it when your type has a sensible default!
impl<Key, Value> Default for SieveCache<Key, Value>
```

**Review by kvc0 — COMMENTED** [review]
looks great - just one doc suggestion

**Comment by swlynch99** [comment]
Doc suggestion merged! I agree that's a good addition.

I think this is all the changes I want to add. So this should be good to go whenever you have time to merge it.

**Review by kvc0 — APPROVED** [review]
Looks great, thank you!

**Pull request merged**